packages/extras/src/lib/components/PerfMonitor/PerfMonitor.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+  import { useStage, useTask, useThrelte, watch } from '@threlte/core'
+  import { ThreePerf } from 'three-perf'
+  import type { PerfMonitorProps } from './PerfMonitor'
+  import { onDestroy } from 'svelte'
+  import { writable } from 'svelte/store'
+
+  type $$Props = PerfMonitorProps
+
+  export let domElement: $$Props['domElement'] = undefined
+  export let logsPerSecond: $$Props['logsPerSecond'] = 10
+  export let showGraph: $$Props['showGraph'] = true
+  export let memory: $$Props['memory'] = true
+  export let enabled: $$Props['enabled'] = true
+  export let visible: $$Props['visible'] = true
+  export let actionToCallUI: $$Props['actionToCallUI'] = ''
+  export let guiVisible: $$Props['guiVisible'] = false
+  export let backgroundOpacity: $$Props['backgroundOpacity'] = 0.7
+  export let scale: $$Props['scale'] = 1
+  export let anchorX: $$Props['anchorX'] = 'left'
+  export let anchorY: $$Props['anchorY'] = 'top'
+
+  const { renderer, renderStage, mainStage } = useThrelte()
+
+  let perf: ThreePerf
+
+  // Has to be fully re-initialized if dom element changes
+  const domElementStore = writable<typeof domElement>(domElement)
+  $: domElementStore.set(domElement)
+  watch([domElementStore], ([domElement]) => {
+    perf = new ThreePerf({
+      anchorX,
+      anchorY,
+      domElement: domElement || document.body,
+      renderer: renderer,
+      logsPerSecond,
+      showGraph,
+      memory,
+      enabled,
+      visible,
+      actionToCallUI,
+      guiVisible,
+      backgroundOpacity,
+      scale
+    })
+  })
+
+  $: perf && (perf.logsPerSecond = logsPerSecond || 10)
+  $: perf && (perf.showGraph = showGraph ? true : false)
+  $: perf && (perf.memory = memory ? true : false)
+  $: perf && (perf.enabled = enabled ? true : false)
+  $: perf && (perf.visible = visible ? true : false)
+  $: perf && (perf.actionToCallUI = actionToCallUI || '')
+  $: perf && (perf.guiVisible = guiVisible ? true : false)
+  $: perf && (perf.backgroundOpacity = backgroundOpacity || 0.7)
+  $: perf && (perf.scale = scale !== undefined ? scale : 1)
+  $: perf && (perf.anchorX = anchorX || 'left')
+  $: perf && (perf.anchorY = anchorY || 'top')
```

Comment:
This can be simplified a bit. First, above if we change the following...

```ts
type $$Props = PerfMonitorProps
type $$PropsWithDefaults = Required<$$Props>

export let domElement: $$PropsWithDefaults['domElement'] = document.body
```

...and set all of the exports to `$$PropsWithDefaults` we can specify that these props will always be defined since they have default values.

Then, here, we can change this block to just the following:

```ts
  $: perf = new ThreePerf({ renderer, domElement })
  $: perf.logsPerSecond = logsPerSecond
  $: perf.showGraph = showGraph
  $: perf.memory = memory
  $: perf.enabled = enabled
  $: perf.visible = visible
  $: perf.actionToCallUI = actionToCallUI
  $: perf.guiVisible = guiVisible
  $: perf.backgroundOpacity = backgroundOpacity
  $: perf.scale = scale
  $: perf.anchorX = anchorX
  $: perf.anchorY = anchorY
```

Since `ThreePerf` only requires those two arguments we can lazily set the rest immediately after and avoid a lot of repetition.